scripts/experimental/test_battery.py

```diff
@@ -1,56 +1,75 @@
-    Ecurrent_li_s=battery_li_s.TotalEnergy
+    battery_li_air                = SUAVE.Components.Energy.Storages.Variable_Mass.Battery_Lithium_Air()
+    battery_al_air                = SUAVE.Components.Energy.Storages.Variable_Mass.Battery_Aluminum_Air()
+    battery_li_air.discharge_model=datta_discharge           #default discharge model, but assign anyway
+    battery_li_ion                = SUAVE.Components.Energy.Storages.Constant_Mass.Battery_Lithium_Ion()
+    battery_li_s                  = SUAVE.Components.Energy.Storages.Constant_Mass.Battery_Lithium_Sulfur()
+    li_ion_mass                   = 10*Units.kg
+    #build numerics
+    numerics.integrate_time       = np.array([[0, 0],[0, 1]])
+    numerics.differentiate_time   = np.array([[0, 0],[0, 1]])
     
-    Ploss_li_s=battery_li_s(Preq,time)
-    print battery_li_s
-    [Ploss_li_air, mdot]=battery_li_air(Preq,time)
-    print battery_li_air
-    print 'mass flow rate=', mdot
- 
-    Ploss_li_ion=battery_li_ion(Preq,time)
-    print battery_li_ion
+    #build battery_inputs
+    battery_inputs.current        =90*Units.amps
+    battery_inputs.power_in       =np.array([[Preq/2.] , [Preq]])
+    battery_inputs.energy_transfer=-np.array([[1000],[2000]])*Units.Wh
     
+    battery_li_ion.inputs=battery_inputs
+    #test_initialize_from_energy_and_power(battery_al_air, Ereq, Preq)
+    #test_mass_gain(battery_al_air)
+    #test_find_ragone_properties(specific_energy_guess,battery_li_s, Ereq,Preq)
+    test_find_ragone_optimum(battery_li_ion,Ereq,Preq)
+    battery_li_ion.current_energy=[[battery_li_ion.max_energy]]
+    battery_li_ion.energy_calc( numerics)
+    print battery_li_ion
+    #test_initialize_from_mass(battery_li_ion,li_ion_mass)
     
-    '''
-    vehicle=SUAVE.Vehicle()
-    fan=SUAVE.Components.Propulsors.Ducted_Fan_Bat()
-    vehicle.battery=battery_li_s
-    fan.battery=battery_li_s
+def test_mass_gain(battery):
+    print battery
+    mass_gain       =find_total_mass_gain(battery)
+    print 'mass_gain=', mass_gain
+    return
+def test_initialize_from_energy_and_power(battery,energy,power):
+    initialize_from_energy_and_power(battery, energy, power)
+    print battery
+    return
+def test_find_ragone_properties(specific_energy,battery,energy,power):
+    find_ragone_properties( specific_energy, battery, energy,power)
+    print battery
+    print 'specific_energy (Wh/kg)=',battery.specific_energy/(Units.Wh/Units.kg)
+    return
+def test_find_ragone_optimum(battery, energy, power):
+    find_ragone_optimum(battery,energy,power)
+    print battery
     
-    #vehicle.battery(1000, 5)
-    print battery_li_s
-    fan.battery(3000,2)
-    print battery_li_s
-    '''
+    print 'specific_energy (Wh/kg)=',battery.specific_energy/(Units.Wh/Units.kg)
+    print 'max_energy [W-h]=', battery.max_energy/Units.Wh
+    return
+def test_initialize_from_mass(battery,mass):
+    initialize_from_mass(battery,mass)
+    print battery
+    return
 if __name__ == '__main__':

```

Comment:
in general this file needs more comments to chunk code and illuminate its structure